### PR TITLE
Remove external-rules feature flag

### PR DESCRIPTION
**Pull request by raulcabello**
## Problem
`external-rules` feature flag was added in Rancher v2.7 and v2.8 to help admins with the transition of the new `RoleTemplate` field `ExternalRules`. That's no longer needed in Rancher v2.9
 
## Solution
Remove `external-rules` feature flag. It will work as it is always enabled, and can't be disabled. 
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Review by pmatseykanets — APPROVED** [review]

File: pkg/rbac/common.go [hunk 1]
```diff
@@ -237,14 +236,16 @@ func RulesFromTemplate(clusterRoles k8srbacv1.ClusterRoleCache, roleTemplates v3
 func gatherRules(clusterRoles k8srbacv1.ClusterRoleCache, roleTemplates v32.RoleTemplateCache, rt *v3.RoleTemplate, rules []rbacv1.PolicyRule, seen map[string]bool) ([]rbacv1.PolicyRule, error) {
 	seen[rt.Name] = true
 
-	if features.ExternalRules.Enabled() && rt.ExternalRules != nil {
-		rules = append(rules, rt.ExternalRules...)
-	} else if rt.External && rt.Context == "cluster" {
-		cr, err := clusterRoles.Get(rt.Name)
-		if err != nil {
-			return nil, err
+	if rt.External {
```
Comment (andreas-kupries) [inline pkg/rbac/common.go:239]: I am not convinced that this change is correct. The original structure is

```
	if features.ExternalRules.Enabled() && rt.ExternalRules != nil {
		rules = append(rules, rt.ExternalRules...)
	} else if rt.External && rt.Context == "cluster" {
		cr, err := clusterRoles.Get(rt.Name)
		if err != nil {
			return nil, err
                }
		rules = append(rules, cr.Rules...)
	}
```

replacing `features.ExternalRules.Enabled()` with `true` yields

```
	if rt.ExternalRules != nil {
		rules = append(rules, rt.ExternalRules...)
	} else if rt.External && rt.Context == "cluster" {
		cr, err := clusterRoles.Get(rt.Name)
		if err != nil {
			return nil, err
                }
		rules = append(rules, cr.Rules...)
	}
```

I do not see how the `rt.External` clause can now be moved outside to become

```
if rt.External {
                if rt.ExternalRules != nil {
			rules = append(rules, rt.ExternalRules...)
		} else if rt.Context == "cluster" {
			cr, err := clusterRoles.Get(rt.Name)
			if err != nil {
				return nil, err
			}
			rules = append(rules, cr.Rules...)
		}
}
```

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
There is a logic change where I do not see how it is correct.

File: pkg/rbac/common.go (hunk 1 above)
Comment (raulcabello) [reply pkg/rbac/common.go:239]: `rt.ExternalRules != nil` implies `external` is true as the webhook enforces that a `RoleTemplate` can't have `ExternalRules` if `external` is not true. see https://github.com/rancher/webhook/blob/release/v0.5/pkg/resources/management.cattle.io/v3/roletemplate/validator.go#L132

That means that:
```
	if rt.ExternalRules != nil {
		rules = append(rules, rt.ExternalRules...)
	} else if rt.External && rt.Context == "cluster" {
		cr, err := clusterRoles.Get(rt.Name)
		if err != nil {
			return nil, err
                }
		rules = append(rules, cr.Rules...)
	}
```
and 

```
if rt.External {
                if rt.ExternalRules != nil {
			rules = append(rules, rt.ExternalRules...)
		} else if rt.Context == "cluster" {
			cr, err := clusterRoles.Get(rt.Name)
			if err != nil {
				return nil, err
			}
			rules = append(rules, cr.Rules...)
		}
}
```
behaves in the same way.  However, the second option is clearer and it'll work even if the webhook changes in the future

Comment (andreas-kupries) [reply pkg/rbac/common.go:239]: Ok, the `rt.ExternalRules != nil implies external is true` is the part I had not known, and with that it makes sense.
Maybe add this as comment.

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**